src/Adapter/Product/CommandHandler/UpdateProductPricesHandler.php

```diff
@@ -51,27 +55,62 @@ final class UpdateProductPricesHandler implements UpdateProductPricesHandlerInte
      */
     private $productPricePropertiesFiller;
 
+    /**
+     * @var ProductSupplierRepository
+     */
+    private $productSupplierRepository;
+
     /**
      * @param ProductRepository $productRepository
      * @param ProductPricePropertiesFiller $productPricePropertiesFiller
      */
     public function __construct(
         ProductRepository $productRepository,
-        ProductPricePropertiesFiller $productPricePropertiesFiller
+        ProductPricePropertiesFiller $productPricePropertiesFiller,
+        ProductSupplierRepository $productSupplierRepository
     ) {
         $this->productRepository = $productRepository;
         $this->productPricePropertiesFiller = $productPricePropertiesFiller;
+        $this->productSupplierRepository = $productSupplierRepository;
     }
 
     /**
      * {@inheritdoc}
      */
     public function handle(UpdateProductPricesCommand $command): void
     {
-        $product = $this->productRepository->get($command->getProductId());
+        if ($command->getShopConstraint()->forAllShops()) {
+            $product = $this->productRepository->get($command->getProductId());
+        } else {
+            $product = $this->productRepository->getForShop($command->getProductId(), $command->getShopConstraint()->getShopId());
```

Comment:
We have a `partialUpdateForShopConstraint` method, why we don't have a `getForShopConstraint` 
```php
$this->productRepository->get($command->getProductId(), $command->getShopConstraint());
```
?

Reply:
Well in the abstract class I added two methods:
- `getObjectModelForShop` (which use a single shop ID)
- `partiallyUpdateObjectModelForShops` (which uses a list of shop ids)

The get is based on only one shop because the override is only related to a single shop (at least for Product, maybe in the future we'll need to add the same thing for ShopGroup). So getting a Product value only makes sense for one shop.

By the way the "regular" `get` without the provided ShopId actually relies on the legacy behaviour and gets the shop ID from context, this part I'm not particularly a fan because it kind of hides some things, but since the constraint for all shops doesn't contain the context shop ID I don't know how to handle this better. One solution could be to rely on the product's default shop ID though. Maybe I should add this in the `get` method.

Reply:
Regarding the partial update method, the method in the abstract class is more generic (it is based on a list of ShopIds) but in the case of the product repository I added a method which relies on ShopConstraint because we have only two use cases:
- Single shopId, then I call `partiallyUpdateObjectModelForShops` with an array of one ShopId
- All shop context, in this case we need to get the list of all associated shop IDs and then call `partiallyUpdateObjectModelForShops` with the appropriate array

So it felt more convenient to use `ShopConstraint` as a parameter since it is accessible in the commands directly, this way the part that gets the list of associated shops is handled internally in the Repository instead of duplicating it in all the handlers.

Reply:
TLDR; 
- updating values by shop constraint makes sense because in the case of AllShop context we can update a bunch of shop at the same time
- but we can't `get` a Product with data for AllShop, nor ShopGroup (it would be possible if our structure of multishop for product was better handled but that's not the case right now) so since it is only possible for single shop context it felt more simple to rely on ShopID directly rather than ShopConstraint

Reply:
> but we can't get a Product with data for AllShop, nor ShopGroup (it would be possible if our structure of multishop for product was better handled but that's not the case right now) so since it is only possible for single shop context it felt more simple to rely on ShopID directly rather than ShopConstraint

If we want, in the future, to harmonize how we get ObjectModel, we could do the following:
```
public function getForShop(int $productId, ShopConstraint $constraint)
{
    if (!$constraint->isSingleShopConstraint()) {
        throw new RuntimeException('Cannot use ShopConstraint All shops or group shops with product entity');
```

This allows someone motivated to provide another implementation - that handles shop groups and all shops so we accept all kind of constraints in the signature, but in the Core implementation we fail if not given the right constraint.

Reply:
Mmm, I get it what you both say makes sense! Then what I'm gonna do is implement `public function getForShop(int $productId, ShopConstraint $constraint)` in the `ProductRepository` (this will make the handler cleaner and the code easier to reuse)

However I will handle this only in the ProductRepository, I don't think we can make it generic in the abstract class (each type of ObjectModel may have different ways of handling the multishop). Instead the asbtract class offer simpler methods (based on shopId or list of shopIds) so that the concrete Repository (here ProdctRepository) can rely on them but it's the concrete class which has the responsibility to interpret the `ShopConstraint`